File: Serene/Serene.Script/Imports/ServiceContracts/Northwind.NoteRow.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    [Imported, Serializable, PreserveMemberCase]
    public partial class NoteRow
    {
        [InlineConstant] public const string IdProperty = "NoteId";
        [InlineConstant] public const string NameProperty = "EntityType";
        [InlineConstant] public const string LocalTextPrefix = "Northwind.Note";
    
        public Int64? NoteId { get; set; }
        public String EntityType { get; set; }
        public Int64? EntityId { get; set; }
        public String Text { get; set; }
        public Int32? InsertUserId { get; set; }
        public String InsertDate { get; set; }
        public String InsertUserDisplayName { get; set; }
    
        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string NoteId = "NoteId";
            [InlineConstant] public const string EntityType = "EntityType";
            [InlineConstant] public const string EntityId = "EntityId";
            [InlineConstant] public const string Text = "Text";
            [InlineConstant] public const string InsertUserId = "InsertUserId";
            [InlineConstant] public const string InsertDate = "InsertDate";
            [InlineConstant] public const string InsertUserDisplayName = "InsertUserDisplayName";
        }
    }
    
}

File: Serene/Serene.Web/Modules/Imports/ServerTypings/Northwind.NoteRow.ts
namespace Serene.Northwind {
    export interface NoteRow {
        NoteId?: number;
        EntityType?: string;
        EntityId?: number;
        Text?: string;
        InsertUserId?: number;
        InsertDate?: string;
        InsertUserDisplayName?: string;
    }

    export namespace NoteRow {
        export const idProperty = 'NoteId';
        export const nameProperty = 'EntityType';
        export const localTextPrefix = 'Northwind.Note';

        export namespace Fields {
            export declare const NoteId: 'NoteId';
            export declare const EntityType: 'EntityType';
            export declare const EntityId: 'EntityId';
            export declare const Text: 'Text';
            export declare const InsertUserId: 'InsertUserId';
            export declare const InsertDate: 'InsertDate';
            export declare const InsertUserDisplayName: 'InsertUserDisplayName';
        }

        ['NoteId', 'EntityType', 'EntityId', 'Text', 'InsertUserId', 'InsertDate', 'InsertUserDisplayName'].forEach(x => (<any>Fields)[x] = x);
    }
}

